csrc/includes/custom_cuda_layers.h

```diff
@@ -22,6 +22,8 @@
 #define MAX_THREAD_ITERATIONS 8  // Maximum 8K
 #define MAX_WARP_NUM 32
 
+#define MAX_REGISTERS 64
```

Comment:
just curious why set the max registers as 64, is it for some special kernels or a common setting? and it looks too small as a max count of register.

Reply:
Interesting question. I think I can push it up, however it is also multiplied by 4, since there is always the loop-unrolling factor of 4. So, it is actually 256. Also, I am worried that if more blocks are assigned to an SM, there won't be enough registers to be allocated and there will be more memory movement.